src/ucp/core/ucp_worker.c

```diff
@@ -831,32 +833,38 @@ static int ucp_worker_iface_find_better(ucp_worker_h worker,
             continue;
         }
 
-        /* Check that another iface:
-         * 1. Supports all capabilities of the target iface (at least), except
-         *    ...CONNECT_TO... caps.
-         * 2. Has the same or better performance charasteristics */
-        if (ucs_test_all_flags(if_iter->attr.cap.flags, test_flags) &&
-            (if_iter->attr.overhead  <= wiface->attr.overhead)      &&
-            (ucp_tl_iface_bandwidth(ctx, &if_iter->attr.bandwidth) >=
-             ucp_tl_iface_bandwidth(ctx, &wiface->attr.bandwidth))  &&
-            (if_iter->attr.priority  >= wiface->attr.priority)) {
-
-            latency_iter = ucp_worker_iface_latency(worker, if_iter);
-            latency_cur  = ucp_worker_iface_latency(worker, wiface);
-            epsilon      = (latency_iter + latency_cur) * 1e-6;
-            if (latency_iter < latency_cur + epsilon) {
-                /* Do not check this iface anymore, because better one exists.
-                 * It helps to avoid the case when two interfaces with the same caps
-                 * and performance exclude each other. */
-                wiface->flags |= UCP_WORKER_IFACE_FLAG_UNUSED;
-                *better_index  = rsc_index;
-                return 1;
-            }
+        latency_iter = ucp_worker_iface_latency(worker, if_iter);
+
+        /* Check that another iface: */
+        if (/* 1. Supports all capabilities of the target iface (at least),
+             *    except ...CONNECT_TO... caps. */
+            ucs_test_all_flags(if_iter->attr.cap.flags, test_flags) &&
+            /* 2. Has the same or better performance characteristics */
+            (if_iter->attr.overhead <= wiface->attr.overhead) &&
+            (ucp_tl_iface_bandwidth(ctx, &if_iter->attr.bandwidth) >= bw_cur) &&
+            (if_iter->attr.priority >= wiface->attr.priority) &&
+            (latency_iter < (latency_cur +
+                             ucp_calc_epsilon(latency_iter, latency_cur))) &&
+            /* 3. The found transport is scalable enough or both
+             *    transport are unscalable */
+            (ucp_is_scalable_transport(ctx, if_iter->attr.max_num_eps) ||
+             !ucp_is_scalable_transport(ctx, wiface->attr.max_num_eps)))
+        {
+            goto out;
         }
     }
 
-    *better_index  = 0;
+    /* Better resource wasn't found */
+    *better_index = 0;
     return 0;
+
+out:
```

Comment:
btw, goto does not look appropriate here, can just do `out` section in place (like it was below)

Reply:
done